dataset/src/test/scala/frameless/functions/NonAggregateFunctionsTests.scala

```diff
@@ -0,0 +1,477 @@
+    }
+
+
+    check(forAll(prop[Int] _))
+    check(forAll(prop[Long] _))
+    check(forAll(prop[Short] _))
+    check(forAll(prop[BigDecimal] _))
+    check(forAll(prop[Byte] _))
+    check(forAll(prop[Double] _))
+  }
+
+
+   /*
+    * Currently not all Collection types play nice with the Encoders.
+    * This test needs to be readressed and Set readded to the Collection Typeclass once these issues are resolved.
+    *
+    * [[https://issues.apache.org/jira/browse/SPARK-18891]]
+    * [[https://issues.apache.org/jira/browse/SPARK-21204]]
+    */
+  test("array_contains"){
+
+
+    val spark = session
+    import spark.implicits._
+
+    val listLength = 10
+    val idxs = Stream.continually(Range(0, listLength)).flatten.toIterator
+
+    abstract class Nth[A, C[A]:CatalystCollection] {
+
+      def nth(c:C[A], idx:Int):A
+    }
+
+    implicit def deriveListNth[A] : Nth[A, List] = new Nth[A, List] {
+      override def nth(c: List[A], idx: Int): A = c(idx)
+    }
+
+    implicit def deriveSeqNth[A] : Nth[A, Seq] = new Nth[A, Seq] {
+      override def nth(c: Seq[A], idx: Int): A = c(idx)
+    }
+
+    implicit def deriveVectorNth[A] : Nth[A, Vector] = new Nth[A, Vector] {
+      override def nth(c: Vector[A], idx: Int): A = c(idx)
+    }
+
+    implicit def deriveArrayNth[A] : Nth[A, Array] = new Nth[A, Array] {
+      override def nth(c: Array[A], idx: Int): A = c(idx)
+    }
+
+
+    def prop[C[_] : CatalystCollection]
+      (
+        values: C[Int],
+        shouldBeIn:Boolean)
+      (
+        implicit nth:Nth[Int, C],
+        encEv: Encoder[C[Int]],
+        tEncEv: TypedEncoder[C[Int]]
+      ) = {
+
+      val contained = if (shouldBeIn) nth.nth(values, idxs.next) else -1
+
+      val cDS = session.createDataset(List(values))
+      val resCompare = cDS
+        .select(org.apache.spark.sql.functions.array_contains(cDS("value"), contained))
+        .map(_.getAs[Boolean](0))
+        .collect().toList
+
+
+      val typedDS = TypedDataset.create(List(X1(values)))
+      val res = typedDS
+        .select(array_contains(typedDS('a), contained))
+        .collect()
+        .run()
+        .toList
+
+      res ?= resCompare
+    }
+
+
+    check(
+      forAll(
+        Gen.listOfN(listLength, Gen.choose(0,100)),
+        Gen.oneOf(true,false)
+      )
+      (prop[List])
+    )
+
+    /*check( Looks like there is no Typed Encoder for Seq type yet
+      forAll(
+        Gen.listOfN(listLength, Gen.choose(0,100)),
+        Gen.oneOf(true,false)
+      )
+      (prop[Seq])
+    )*/
+
+    check(
+      forAll(
+        Gen.listOfN(listLength, Gen.choose(0,100)).map(_.toVector),
+        Gen.oneOf(true,false)
+      )
+      (prop[Vector])
+    )
+
+    check(
+      forAll(
+        Gen.listOfN(listLength, Gen.choose(0,100)).map(_.toArray),
+        Gen.oneOf(true,false)
+      )
+      (prop[Array])
+    )
+  }
+
+  test("ascii"){
+    val spark = session
+    import spark.implicits._
+
+    def prop(values:List[String]) = {
```

Comment:
In these tests you could make prop to taka `List[X1[String]]` so you don't have to do a lot of extra work, like use the "values" field, and then having to `map(X1(_))`.

Reply:
This holds for many of the tests herein.